Polymorphine Fix

Conga0 opened the pull request:
Fixes most TI votes to respect polymorphine
Fixes dryspell allowing liquid material damage while active

Conga0 committed:
Poly Fix

Fixes most TI votes to respect polymorphine
Fixes dryspell allowing liquid material damage while active

Conga0 committed:
Manifest update

:')

Miczu commented:
![image](https://github.com/user-attachments/assets/2ddcfe20-fc32-491a-b029-596c3cdd8521)

Conga0 committed:
Update manifest.json

Conga0 committed:
Update manifest.json

Miczu commented:
![image](https://github.com/user-attachments/assets/b574333a-25b3-4c67-a717-a435e84dfc71)
wait(1) probably isn't legal to be used in this way 😢 
Become_speed and big_confuse don't work and that's the first 2 I tested.

Conga0 committed:
Update big_confuse.lua

Conga0 committed:
Update manifest.json

Conga0 committed:
New Async

Conga0 committed:
Update manifest.json

Miczu commented:
Game is crashing now when player isn't around - both when poly is user or when I teleport with dev mode that removes player

Conga0 committed:
get_player() nil and cut health wait

get_player() returns nil and cut health waits 30 frames.
20 frames appears to be the poly invincibility window from testing but I sett it to 10 to be safe

Conga0 committed:
Update manifest.json

Miczu commented:
**Forced Checkpoint throws an error** during polymorph, works fine as mina.
![image](https://github.com/user-attachments/assets/867790a3-da1e-4969-9694-faf99a7dd2fb)

**Labyrinth doesn't work** during polymorph and works fine as mine - no console error (hidden by async probably).

Anything using get_player_nopoly just worked, so got kind of lazy and didn't test all of them.

Conga0 committed:
Utils update

Fixed utils.lua to use the correct polymorphed player tag

Conga0 committed:
Update manifest.json

Conga0 committed:
Merge branch 'master' into master

Conga0 commented:
I remember reading online that player_polymorphed was the right tag but it seems I got it backwards, polymorphed_player is what the game actually gives polymorphed players.
Now it should properly detect players, polymorphed players and cessated players.

Conga0 committed:
Update utils.lua

Conga0 committed:
Merge branch 'master' of https://github.com/Conga0/Noita-Twitch-Integration

Conga0 committed:
Update manifest.json

Conga0 committed:
New function

Conga0 committed:
Update manifest.json

Conga0 commented:
Outcomes should now use get_player_event() in place of get_player(), get_player has been returned to it's current live behavior

Miczu reviewed (approved):
Great fix, works as expected! 
Hmm... get_player_event... your function naming skills could use some work.
